coderadar-graph/src/main/java/io/reflectoring/coderadar/graph/query/adapter/GetCriticalFilesAdapter.java

```diff
@@ -1,28 +1,65 @@
 package io.reflectoring.coderadar.graph.query.adapter;
 
+import io.reflectoring.coderadar.graph.analyzer.repository.FileRepository;
+import io.reflectoring.coderadar.graph.projectadministration.project.adapter.CommitBaseDataMapper;
 import io.reflectoring.coderadar.graph.query.domain.ContributorsForFileQueryResult;
+import io.reflectoring.coderadar.graph.query.domain.FileAndCommitsForTimePeriodQueryResult;
 import io.reflectoring.coderadar.graph.query.repository.ContributorQueryRepository;
 import io.reflectoring.coderadar.projectadministration.domain.FilePattern;
 import io.reflectoring.coderadar.projectadministration.domain.InclusionType;
 import io.reflectoring.coderadar.query.domain.ContributorsForFile;
+import io.reflectoring.coderadar.query.domain.FileAndCommitsForTimePeriod;
 import io.reflectoring.coderadar.query.port.driven.GetCriticalFilesPort;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
+import org.springframework.data.util.Pair;
 import org.springframework.stereotype.Service;
 
 @Service
 public class GetCriticalFilesAdapter implements GetCriticalFilesPort {
   private final ContributorQueryRepository contributorQueryRepository;
+  private final FileRepository fileRepository;
+  private final CommitBaseDataMapper mapper = new CommitBaseDataMapper();
 
-  public GetCriticalFilesAdapter(ContributorQueryRepository contributorQueryRepository) {
+  public GetCriticalFilesAdapter(
+      ContributorQueryRepository contributorQueryRepository, FileRepository fileRepository) {
     this.contributorQueryRepository = contributorQueryRepository;
+    this.fileRepository = fileRepository;
   }
 
   @Override
-  public List<ContributorsForFile> getCriticalFiles(
-      Long projectId, int numberOfContributors, String commitHash, List<FilePattern> filePatterns) {
+  public List<ContributorsForFile> getFilesWithManyContributors(
+      long projectId, int numberOfContributors, String commitHash, List<FilePattern> filePatterns) {
+    Pair<List<String>, List<String>> includesAndExcludes = mapPatternsToRegex(filePatterns);
+    return mapContributorsForFileResult(
+        contributorQueryRepository.getFilesWithManyContributors(
+            projectId,
+            numberOfContributors,
+            commitHash,
+            includesAndExcludes.getFirst(),
+            includesAndExcludes.getSecond()));
+  }
+
+  @Override
+  public List<FileAndCommitsForTimePeriod> getFrequentlyChangedFiles(
```

Comment:
What happens if the start date is after the commit time?

Reply:
Nothing is returned. The date of the commits need to be at a later time than start date. We could declare this as a misconfiguration and tell the user that the input is invalid.